cdap-watchdog/src/main/java/co/cask/cdap/metrics/process/MessagingMetricsProcessorService.java

```diff
@@ -302,8 +329,9 @@ private void processMetrics() {
             RawMessage input = iterator.next();
             try {
               payloadInput.reset(input.getPayload());
-              MetricValues metricValues = recordReader.read(decoder, recordSchema);
-              records.add(metricValues);
+              MetricValues metricValues = metricReader.read(decoder, metricSchema);
+              metricsFromAllTopics.put(metricValues);
+              lastMetricTimeMillis = TimeUnit.SECONDS.toMillis(metricValues.getTimestamp());
```

Comment:
You can instead keep the `lastMetricTimeSecs` to avoid conversion inside the loop.